xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -171,23 +181,21 @@ int GetDisplayIndex(CGDirectDisplayID display)
   return -1;
 }
 
-int GetDisplayIndex(const std::string& dispName)
+NSUInteger GetDisplayIndex(const std::string& dispName)
 {
-  int ret = 0;
+  NSUInteger ret = 0;
 
   // Add full screen settings for additional monitors
-  int numDisplays = NSScreen.screens.count;
-
-  for (int disp = 0; disp < numDisplays; disp++)
+  const NSUInteger numDisplays = NSScreen.screens.count;
+  for (NSUInteger disp = 0; disp <= numDisplays - 1; disp++)
```

Comment:
you might get funny results when `numDisplays == 0`, might be possible if Kodi is started with screen off. There're also other such places in the PR.

Reply:
This code was already there, I've no way of checking this as I don't have any mac without a builtin display. Hence, I prefer not to touch it. Maybe someone with the required hardware may give it a whirl with no display later

Reply:
existing code uses signed integer, and you've changed it to unsigned. `0 - 1` as unsigned won't be `-1` ;)

Reply:
I don't think it'll make a difference as we append "Default" by default, so the list of available screens is always > 0. Let's try to catch any misbehaviour later on if someone complains.